[DESCW-2991] Add DatabaseConfig::getConnectionArray()

mhaswell-bcgov opened the pull request:
Update Database class and migrations-db script to use above
Fix typo in NaadSocketClientTest function

ShawnTurple reviewed (changes requested):
Looks good, although we should update the version, and assuming we are going to use releases, we don't need to update the changelog, and that information will be obvious in the release.

mhaswell-bcgov commented:
> Looks good, although we should update the version, and assuming we are going to use releases, we don't need to update the changelog, and that information will be obvious in the release.

@ShawnTurple I removed the version and added a license to composer.json per `composer validate` recommendations:
```
# General warnings
- No license specified, it is recommended to do so. For closed-source software you may use "proprietary" as license.
- The version field is present, it is recommended to leave it out if the package is published on Packagist.
```

ShawnTurple commented on composer.json:
```diff
@@ -1,8 +1,8 @@
 {
   "name": "bcgov/naad-connector",
-  "version": "1.0.0",
```
why did you remove this?

mhaswell-bcgov replied:
It serves no purpose in this project since we're not installing this project anywhere via composer. When you run `composer validate` it actually flags having the `version` field at all as a warning:
```
- The version field is present, it is recommended to leave it out if the package is published on Packagist.
```

ShawnTurple replied:
ok, so then how are we keeping track of versions, agree that in this case it is not used, but when we are developing things, how is the version going to be tracked, if you say simply tagging, then i would say we have a few processes to work out.

ShawnTurple reviewed (approved):
i'm ok approving this, but i suspect there are still some process things we need to figure out with versions
